sigopt/xgboost/experiment.py

```diff
@@ -0,0 +1,82 @@
+from .run import run as XGBRun
+from .run import parse_run_options, PARAMS_LOGGED_AS_METADATA, DEFAULT_EVALS_NAME
+import copy
+
+from .. import create_experiment
+
+DEFAULT_CLASSIFICATION_METRICS = ['accuracy', 'precision', 'recall', 'F1']
+DEFAULT_REGRESSION_METRICS = ['mean absolute error', 'mean squared error']
+DEFAULT_NUM_BOOST_ROUND = 10
+
+
+class XGBExperiment:
+  def __init__(self, experiment_config, dtrain, evals, params, num_boost_round, run_options):
+    self.experiment_config = experiment_config
+    self.dtrain = dtrain
+    self.evals = evals
+    self.params = params
+    self.num_boost_round = num_boost_round
+    self.run_options = run_options
+    self.sigopt_experiment = None
+
+  def parse_and_create_parameter_space(self):
+    # TODO: deal with later
+    pass
+
+  def parse_and_create_experiment(self):
+    experiment_config_parsed = copy.deepcopy(self.experiment_config)
+
+    # Check experiment config optimization metric
+    for metric in experiment_config_parsed['metrics']:
+      if metric['strategy'] == 'optimize':
+        assert metric['name'] in DEFAULT_CLASSIFICATION_METRICS or metric['name'] in DEFAULT_REGRESSION_METRICS
+
+        # change optimized metric to reflect updated name
+        metric['name'] = self.evals[0][1] + '-' + metric['name'] if isinstance(self.evals, list) else \
+          DEFAULT_EVALS_NAME + '-' + metric['name']
+
+    # Check key overlap between parameters to be optimized and parameters that are set
+    params_optimized = [param['name'] for param in self.experiment_config['parameters']]
+    assert len(set(params_optimized) & set(self.params.keys())) == 0, \
+      'There is overlap between optimized params and user-set params'
```

Comment:
`There is overlap between tuned parameters (through `experiment_config`) and user-set parameters (through `params`). Parameter name cannot be defined in both locations.` 

Ideally, I want this message to be even more informative, like listing out the parameter names that are overlapping. Don't have to have it right now. But can you add it as a TODO comment  here?